lib/internal/url.js

```diff
@@ -77,7 +78,7 @@ const {
   CHAR_PERCENT,
   CHAR_PLUS,
 } = require('internal/constants');
-const path = require('path');
+const path = getLazy(() => require('path'));
```

Comment:
```suggestion
const lazyPath = getLazy(() => require('path'));
```

Reply:
Why did we make this change?

Reply:
The lazy loading prevents a circular requirement between the path and url implementations